hid_enabled.go

```diff
@@ -226,3 +359,39 @@ func (dev *Device) Read(b []byte) (int, error) {
 	}
 	return read, nil
 }
+
+// Type identify the device as a HID device
+func (dev *HidDevice) Type() DeviceType {
+	return dev.DeviceInfo.Type()
+}
+
+// GenericDeviceHandle represents a libusb device_handle struct
+type GenericDeviceHandle *C.struct_libusb_device_handle
+
+// GenericLibUsbDevice represents a libusb device struct
+type GenericLibUsbDevice *C.struct_libusb_device
```

Comment:
This type, `GenericDeviceHandle` and the associated functions should not be exported. Exporting C types is discouraged in Go bindings for two reasons: these exports will not be present when cgo is unavailable and C types often have special rules regarding their lifetime.